pkg/guacanalytics/patchPlanning_test.go

```diff
@@ -0,0 +1,399 @@
+		expectedLen    int
+		graphInput     string
+	}{
+
+		{
+			name:           "1: test case with two dependencies at the same depth, no stopID and no limiting maxDepth",
+			startType:      "deb",
+			startNamespace: ptrfrom.String("ubuntu"),
+			startName:      "dpkg",
+			stopType:       "",
+			maxDepth:       10,
+			expectedLen:    3,
+			graphInput:     "isDependency",
+		},
+		{
+			name:           "2: two levels of dependencies, no stopID and no limiting maxDepth",
+			startType:      "top",
+			startNamespace: ptrfrom.String("topns"),
+			startName:      "toppkg",
+			stopType:       "",
+			maxDepth:       10,
+			expectedLen:    4,
+			graphInput:     "isDependency",
+		},
+
+		{
+			name:           "3: two levels of dependencies, a stopID at the first level and no limiting maxDepth",
+			startType:      "top",
+			startNamespace: ptrfrom.String("topns"),
+			startName:      "toppkg",
+			stopType:       "deb",
+			stopNamespace:  ptrfrom.String("ubuntu"),
+			stopName:       "dpkg",
+			maxDepth:       10,
+			expectedLen:    2,
+			graphInput:     "isDependency",
+		},
+
+		{
+			name:           "4: two levels of dependencies, no stopID and a limiting maxDepth at the first level",
+			startType:      "top1",
+			startNamespace: ptrfrom.String("topns1"),
+			startName:      "toppkg1",
+			stopType:       "",
+			maxDepth:       1,
+			expectedLen:    2,
+			graphInput:     "isDependency",
+		},
+		{
+			name:           "5: isDependency indirect dependency",
+			startType:      "top2",
+			startNamespace: ptrfrom.String("topns2"),
+			startName:      "toppkg2",
+			stopType:       "",
+			maxDepth:       10,
+			expectedLen:    5,
+			graphInput:     "isDependency",
+		},
+		{
+			name:           "6: isDependency range that does not include the dependency",
+			startType:      "top3",
+			startNamespace: ptrfrom.String("topns3"),
+			startName:      "toppkg3",
+			stopType:       "",
+			maxDepth:       10,
+			expectedLen:    1,
+			graphInput:     "isDependency",
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(fmt.Sprintf("Test case %s\n", tt.name), func(t *testing.T) {
+			ingestTestData(tt.graphInput, ctx, gqlclient)
+			startID := getPackageId("isDependency", tt.startType, tt.startNamespace, tt.startName, ctx, gqlclient)
+
+			var stopID string
+			if tt.stopType != "" {
+				stopID = getPackageId("isDependency", tt.stopType, tt.stopNamespace, tt.stopName, ctx, gqlclient)
+			} else {
+				stopID = ""
+			}
+
+			gotMap, err := SearchDependenciesFromStartNode(ctx, gqlclient, startID, stopID, "packageVersion", tt.maxDepth)
+
+			if err != nil {
+				t.Errorf("got err from SearchDependenciesFromStartNode: %v", err)
+				return
+			}
+
+			if stopID == "" && tt.maxDepth == 10 {
+				for k, v := range gotMap {
+					if !v.expanded {
+						t.Errorf("All nodes should be expanded but this node was not: node %s \n", k)
+					}
+				}
+			}
+
+			if diff := cmp.Diff(tt.expectedLen, len(gotMap)); len(diff) > 0 {
```

Comment:
can you compare a sorted list? To make sure we get the path we expect back when searching for a node

Reply:
I added an expected outputs field to resolve this.